device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -842,7 +856,11 @@ public void onSessionClosedUnexpectedly(ErrorCondition errorCondition, String de
         // If the session closes during an open call, need to decrement this latch so that the open call doesn't wait
         // for this session to open. The above call to onMultiplexedDeviceSessionRegistrationFailed will report
         // the relevant exception.
-        this.deviceSessionsOpenedLatches.get(deviceId).countDown();
+        CountDownLatch deviceSessionOpenedLatch = this.deviceSessionsOpenedLatches.get(deviceId);
+        if (deviceSessionOpenedLatch != null)
+        {
+            deviceSessionOpenedLatch.countDown();
```

Comment:
Adding a null check here just in case a client is unregistered from the multiplexing client during open somehow (this latch would be null in that case, and we wouldn't need to wait on it anymore)